### ISSUE TITLE
"%3A" as destination in 'redirects' of vercel.json = Invalid route destination segment

### ISSUE THREAD

**Issue by wiiiimm** [zeit/now#10841]
# TLDR;

1. The escaped `%3A` should not be treated as `:` and as the start of a named segment in . 
2. `next.config.js` has it _right_. 
3. `vercel.json` has it _wrong_.

# Problem

`%3A` is `:` URL encoded.

When you have `%3A` in your **redirect** (and likely **rewrites**) destination of `vercel.json` (but not `next.config`), Vercel thinks there is a named segment involves in the destination and will throw a [Invalid route destination segment](https://vercel.com/docs/errors/error-list#invalid-route-destination-segment).

```
// vercel.json
{
  "redirects": [
    {
      "source": "/vercel",
      "destination": "https://www.google.com/search?q=vercel%3Adeploy",
      "permanent": false
    }
  ]
}
```

This is not an issue with `next.config.ts` or `next.config.js`. The following is _valid_:
```
/** @type {import('next').NextConfig} */
const nextConfig = {
  async redirects() {
    return [
      {
        source: '/vercel',
        destination: 'https://www.google.com/search?q=vercel%3Adeploy',
        permanent: false,
      }
    ]
  },
}

module.exports = nextConfig
```
